crud.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from sqlalchemy.orm import Session
+
+import schemas
+from db import models
+
+
+def get_all_authors_paginated(db: Session, page: int = 0, size: int = 10):
+    offset = page * size
+    queryset = db.query(models.Author).offset(offset).limit(size)
+    return queryset.all()
+
+
+def get_all_books_paginated(
+    db: Session, author_id: Optional[int] = None, page: int = 0, size: int = 10
+):
+    offset = page * size
+    queryset = db.query(models.Book)
+    if author_id is not None:
+        queryset = queryset.filter(models.Book.author_id == author_id)
+    return queryset.offset(offset).limit(size).all()
+
+
+def get_author_by_id(db: Session, author_id: int):
+    return db.query(
+        models.Author
+    ).filter(
+        models.Author.id == author_id
+    ).first()
+
+
+def get_author_by_name(db: Session, name: str):
+    return db.query(models.Author).filter(models.Author.name == name).first()
+
+
+def create_author(db: Session, author: schemas.AuthorCreate):
+    new_author = models.Author(**author.model_dump())
```

Comment:
Ensure that `schemas.AuthorCreate` has a `model_dump()` method or replace it with the correct method to convert the schema object to a dictionary. This is crucial for the unpacking operation to work correctly.